Common/tests/Auth/AuthenticationIntegrationTest.cs

```diff
@@ -177,38 +178,76 @@ public AuthenticationIntegrationTest(AuthenticationType type)
     TestContext.Progress.WriteLine(options_.ImpersonationUsernameHeader);
   }
 
+  /// <summary>
+  ///   Enum specifying the index of the different user cases
+  /// </summary>
   public enum IdentityIndex
   {
+    // Request has been sent without headers
     MissingHeaders = -2,
-    DoesntExist    = -1,
-    AllRights      = 0,
-    NoRights       = 1,
+
+    // The user doesn't exist
+    DoesntExist = -1,
+
+    // The user has all permissions
+    AllRights = 0,
+
+    // The user has no permissions
+    NoRights = 1,
+
+    // The user has the permission to impersonate
     CanImpersonate = 2,
-    NoCertificate  = 3,
-    SomeRights     = 4,
-    OtherRights    = 5,
+
+    // The user has no valid certificate
+    NoCertificate = 3,
+
+    // The user has half of the permissions
+    SomeRights = 4,
+
+    // The user has the other half of the permissions
+    OtherRights = 5,
   }
 
+  /// <summary>
+  ///   Defines the expected behavior
+  /// </summary>
   public enum ResultType
   {
+    // User is always authorized
     AlwaysTrue,
+
+    //User is never authorized
     AlwaysFalse,
+
+    //User is only authorized for the permissions they have
```

Comment:
```suggestion
    // User is never authorized
    AlwaysFalse,

    // User is only authorized for the permissions they have
```

Could it work to have xml docs `///` ?